### ISSUE TITLE
Pressing `Ctrl+C` when `--inspector-brk` is used but debugger is not connected hangs Vitest

### ISSUE THREAD

**Issue by sheremet-va** [vitest-dev/vitest#5511]
### Describe the bug

When trying to debug Vitest with a break, Vitest hangs indefinitely if no connection is ever established.

### Reproduction

```shell
vitest --inspect-brk --no-file-parallelism
Ctrl+C
```

https://stackblitz.com/fork/github/vitest-dev/vitest/tree/main/examples/basic?initialPath=__vitest__/

### System Info

```shell
Any
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by AriPerkkio** [comment]
This is what's happening when `CTRL+c` is pressed:

```js
import { isMainThread, Worker } from "node:worker_threads";
import { fileURLToPath } from "node:url";
import inspector from "node:inspector";

const filename = fileURLToPath(import.meta.url);

if (isMainThread) {
  const worker = new Worker(filename);

  await new Promise((r) => setTimeout(r, 1_000));

  console.log("Terminating...");
  const timeout = setTimeout(() => console.log("Stuck here"), 2_000);
  await worker.terminate();

  clearTimeout(timeout);
  console.log("Done");
} else {
  inspector.open();
  inspector.waitForDebugger();
}
```

```
$ node worker.mjs 
Debugger listening on ws://127.0.0.1:9229/65cec5ee-87d3-4a0c-b481-ac9d257a96f5
For help, see: https://nodejs.org/en/docs/inspector
Terminating...
Stuck here
```

**Comment by sheremet-va** [comment]
Maybe we can send the event `Runtime.runIfWaitingForDebugger` somehow before terminating 🤔

**Comment by AriPerkkio** [comment]
For that I think we would need to connect to the Worker's inspector session from main thread with web socket. There's also [`session.connectToMainThread()`](https://nodejs.org/api/inspector.html#sessionconnecttomainthread_1) but I've never got it working and there are no examples for it anywhere. 

For the inspector session's web socket connection we have similar setups here:

https://github.com/vitest-dev/vitest/blob/6157282cc04494afc85085775d1f565c0d0dbb5a/test/inspect/test/inspect.test.ts#L13-L16

Worker not terminating when it's paused in `waitForDebugger()` sounds like a Node bug to me. Though it's still somehow able to terminate when default `CTRL+c` happens and `process.stdin` raw mode is not overwritten. 🤔

**Comment by sheremet-va** [comment]
> For that I think we would need to connect to the Worker's inspector session from main thread with web socket.

Can't we send `postMessage` from the main thread to the worker where the inspector is already initiated?

**Comment by AriPerkkio** [comment]
Inside the Worker `setTimeout` should be blocked when `waitForDebugger()` is active. I don't think Worker receives anything from `postMessage` either.

**Comment by AriPerkkio** [comment]
This should be fixed in Node v22.2.0: https://github.com/nodejs/node/pull/52971
